From d779f686f3fdcdecef26a6ab0db6fd5c1e63dce8 Mon Sep 17 00:00:00 2001
From: Luca Magrone <luca@magrone.cc>
Date: Fri, 4 Nov 2022 15:52:00 +0100
Subject: [PATCH] Use btattach instead of hciattach

Use btattach command to attach bluetooth uart instead of the old
hciattach which on latest RHEL/Fedora fails to load the firmware

Signed-off-by: Luca Magrone <luca@magrone.cc>
---
 usr/bin/btuart | 19 +++++--------------
 1 file changed, 5 insertions(+), 14 deletions(-)

diff --git a/usr/bin/btuart b/usr/bin/btuart
index 493fd86..3c137e0 100755
--- a/usr/bin/btuart
+++ b/usr/bin/btuart
@@ -1,16 +1,7 @@
 #!/bin/sh
 
-HCIATTACH=/usr/bin/hciattach
-if grep -q "raspberrypi,4" /proc/device-tree/compatible; then
-  BDADDR=
-else
-  SERIAL=`cat /proc/device-tree/serial-number | cut -c9-`
-  B1=`echo $SERIAL | cut -c3-4`
-  B2=`echo $SERIAL | cut -c5-6`
-  B3=`echo $SERIAL | cut -c7-8`
-  BDADDR=`printf b8:27:eb:%02x:%02x:%02x $((0x$B1 ^ 0xaa)) $((0x$B2 ^ 0xaa)) $((0x$B3 ^ 0xaa))`
-fi
-
+BTATTACH=/usr/bin/btattach
+#
 # Bail out if the kernel is managing the Bluetooth modem initialisation
 if ( dmesg | grep -q -E "hci[0-9]+: BCM: chip" ); then
   # On-board bluetooth is already enabled
@@ -23,10 +14,10 @@ serial1="`cat /proc/device-tree/aliases/serial1`"
 if [ "$uart0" = "$serial1" ] ; then
 	uart0_pins="`wc -c /proc/device-tree/soc/gpio@7e200000/uart0_pins/brcm\,pins | cut -f 1 -d ' '`"
 	if [ "$uart0_pins" = "16" ] ; then
-		$HCIATTACH /dev/serial1 bcm43xx 3000000 flow - $BDADDR
+		$BTATTACH -B /dev/serial1 -P bcm -S 3000000 &
 	else
-		$HCIATTACH /dev/serial1 bcm43xx 921600 noflow - $BDADDR
+		$BTATTACH -B /dev/serial1 -P bcm -S  921600 -N &
 	fi
 else
-	$HCIATTACH /dev/serial1 bcm43xx 460800 noflow - $BDADDR
+	$BTATTACH -B /dev/serial1 -P bcm -S 460800 -N &
 fi
-- 
2.38.1

